src/main/java/seedu/address/logic/commands/MailXCommand.java

```diff
@@ -0,0 +1,279 @@
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+
+/**
+ * Opens the system default mail app with people
+ * identified by the arguments specified by the user.
+ */
+public class MailXCommand extends Command {
+
+    public static final String COMMAND_WORD = "mail-x";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Opens the mail box with the emails of the people identified by parameters \n"
+            + "Optionally include parameters: \n"
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_GROUP + "GROUP] "
+            + "[" + PREFIX_MOD + "MOD] "
+            + "Example: " + COMMAND_WORD + " 1"
+            + "or " + COMMAND_WORD + PREFIX_EMAIL + "johndoe@example.com";
+
+    public static final String MESSAGE_SUCCESS = "Opening System Default Mail App";
+
+    public static final String NAME_NOT_PRESENT = "Name not present in the contact list";
+
+    public static final String EMAIL_NOT_PRESENT = "Email not present in the contact list";
+
+    public static final String MOD_NOT_PRESENT = "Module not present in the contact list";
+
+    public static final String GROUP_NOT_PRESENT = "Group not present in the contact list";
+
+    private final MailMDescriptor mailMDescriptor;
+
+    /**
+     * Constructor for the MailXCommand class.
+     * @param mailMDescriptor The {@code MailMDescriptor} object which collates all the email addresses to mail to.
+     */
+    public MailXCommand(MailMDescriptor mailMDescriptor) {
+        requireNonNull(mailMDescriptor);
+
+        this.mailMDescriptor = mailMDescriptor;
+    }
+
+    /**
+     * Executes the mail-m command.
+     * @param model {@code Model} which the command should operate on.
+     * @return {@code CommandResult} object that contains the result of the execution.
+     * @throws CommandException if any noncompliance occurs in the command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        launchMail(createEmailList(this.mailMDescriptor, model));
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    /**
+     * Creates the email list as specified the arguments passed by the user.
+     * @param mailMDescriptor The {@code MailMDescriptor} object which collates all the email addresses to mail to.
+     * @param model {@code Model} which the command should operate on.
+     * @return The list of email addresses to whom the mail is directed towards.
+     * @throws CommandException if the command execution goes unexpected.
+     */
+    private Email[] createEmailList(MailMDescriptor mailMDescriptor, Model model) throws CommandException {
+        List<Email> emailList = new ArrayList<>();
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Name name = mailMDescriptor.getName().orElse(null);
+        if (!isNull(name)) {
```

Comment:
A nit, but abstracting these as boolean conditions separately instead of inside the if() would make the code much more readable. These conditions could be `&&` together to reduce nesting as well.

Reply:
Okay, on it!

Reply:
Fixed! You may have a look.